context-free.md

```diff
@@ -1,21 +1,39 @@
 # Context Free
 
 ##  Who is this programming language for?
-
+This programming language is used to create images, or more specifically, create complex images using only variations of
+	circles, triangles, and squares. Combining with the ability to render images with size up to 100 mega pixels, I believe 
+	the program is directed towards people who liked to design but just not o nthe professional level yet. 
 
 ## What is easy to do in this language? Why is it easy?
-
+Like said above, this program has the ability to manipulate a large amount of shapes, allowing people to create images 
+	only defined by their coding ability and tehir imagination. Also, as a DSL, Context Free excels in handling programs 
+	written to create images only and also by using precise "expressions" that is specific to this area. 
 
 ## What is hard to do in this language? Why is it hard?
-
+Due to the properties of Context Free and DSLs, their scope of use is relatively narrow, many are designed specificlly to
+	complete directions specific for one category. This meant that it is hard for Context Free to complete things outsisw of 
+	its scope. One challenge when using Context Free is that the program shapes is limited to circles, squares, and triangles,
+	which meant that it is hard for one to draw something like a detailed car, or other things with irregular shapes. In short,
+	the program excels in completing the tasks it is designed for, but not for completing other actions.
 
 ## How did you learn how to program in this language?
-
+I learned how to program in this language by using the github wiki on Context Free to learn about the language, uses of each
+	codem and also the appropriate time to use it. I also took ideas and also concepts from the example program to strengthen my
+	understanding.
 
 ## Describe what happens when a ContextFree program runs
 _Specifically, how does the computer interpret the rules in a program, in
 order to turn those rules into a picture?_
+When a Context Free program runs, the program is similar to a tree. The program takes the basic shapes of cirle, square, and 
+	triangle to create complex shapes. The complex shapes are difined and created by the "rules." Each rule is basically how to 
+	"build" each shape. By expnding the most complex shapes into shaoes smaller and smaller and eventually into the basic shapes
+	the program is able to create complex images with simple ideas and little code. 
 
 
 ## What do you think is interesting about the ContextFree program you wrote?
+I believe the most interesting thing about the program I wrote is the layout and the ordered structure of the image. Before, 
```

Comment:
I like your picture. I was surprised to see that you can generate such a tidy picture with context-free. My picture was completely chaos comparing to yours. 
I actually learned from your codes in this section, where you multiply the ```PART``` rule by 9, and rotate after each multiplication.
```
9* {r 80} PART {}
```
I really don't know that I can multiply the rules directly, while rotating the shapes after each iteration. You really made a nice picture with nice coding!